App.js

```diff
@@ -1,123 +1,219 @@
+      RNSentiance.listenUserActivityUpdates();
+    }
+  }
+
+  async onSdkStatusUpdate(sdkStatus) {
+    const data = await this.statusToData(sdkStatus);
+    this.setState({
+      data: data
+    });
+  }
+
+  onUserActivityUpdate(userActivity) {
+    const { type, stationaryInfo } = userActivity;
+    let userActivityText = "";
+
+    if (type === "USER_ACTIVITY_TYPE_STATIONARY") {
+      const { location } = stationaryInfo;
+      if (location) {
+        const { latitude, longitude } = location;
+        stationaryLocation = `${parseFloat(latitude)},${parseFloat(longitude)}`;
+      }
+      userActivityText = `Stationary @${stationaryLocation}`;
+    } else if (type === "USER_ACTIVITY_TYPE_TRIP") {
+      userActivityText = "Trip";
+    } else if (type === "USER_ACTIVITY_TYPE_UNKNOWN") {
+      userActivityText = "Unknown";
+    }
+    this.setState({
+      userActivityText
+    });
+  }
+
+  async isSdkInitialized() {
+    const initState = await RNSentiance.getInitState();
+    return initState == "INITIALIZED";
+  }
+
+  async shouldInitialize() {
+    const initState = await RNSentiance.getInitState();
+    return initState == "NOT_INITIALIZED";
+  }
+
+  copyUserIdToBuffer = () => {
+    Clipboard.setString(this.state.userId);
   };
-  setSDKProperties(sdkProperties);
-};
-
-const App = () => {
-  const [sdkProperties, setSDKProperties] = useState({});
-  // Run an effect only after the firsrt render to fetch SDK properties
-  useEffect(() => {
-    getSDKPropertiesFromBridge(setSDKProperties);
-  }, []);
-  return (
-    <Fragment>
-      <StatusBar barStyle="dark-content" />
-      <SafeAreaView>
-        <ScrollView
-          contentInsetAdjustmentBehavior="automatic"
-          style={styles.scrollView}
+
+  async requestPermissionAndroid() {
+    if (Platform.OS === "android") {
+      const granted = await PermissionsAndroid.request(
+        PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION
+      );
+      if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+        const sdkInitialized = await this.isSdkInitialized();
+        if (sdkInitialized) {
+          const sdkStatus = await RNSentiance.getSdkStatus();
+          this.onSdkStatusUpdate(sdkStatus);
+        }
+      }
+    }
+  }
+
+  async statusToData(sdkStatus) {
+    const {
+      startStatus,
+      isRemoteEnabled,
+      isLocationPermGranted,
+      locationSetting,
+      isAccelPresent,
+      isGyroPresent,
+      isGooglePlayServicesMissing,
+      wifiQuotaStatus,
+      mobileQuotaStatus,
+      diskQuotaStatus
+    } = sdkStatus;
+
+    const diskQuota = await RNSentiance.getDiskQuotaLimit();
+    const diskQuotaUsed = await RNSentiance.getDiskQuotaUsage();
+    const mobileQuota = await RNSentiance.getMobileQuotaLimit();
+    const mobileQuotaUsed = await RNSentiance.getMobileQuotaUsage();
+    const wifiQuota = await RNSentiance.getWiFiQuotaLimit();
+    const wifiQuotaUsed = await RNSentiance.getWiFiQuotaUsage();
+
+    data = [
```

Comment:
Why not just export an object instead of manually creating key/value pairs?

Reply:
While ES6 / ES2015 adds property order, it does not require for-in or Object.keys to follow that order, due to legacy compatibility concerns. However we could do it without any issues. Think array is a bit more natural here though.